spec-0007/transition_to_rng.py

```diff
@@ -146,11 +160,34 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-# Example usage of _prepare_rng decorator
-@_transition_to_rng("random_state", 1, "1.15.0")
-# previously, the signature of the function was
-#   library_function(arg1, random_state=None, arg2=None):
-def library_function(arg1, rng=None, arg2=None):
-    if isinstance(rng, (np.random.Generator, np.random.RandomState)):
-        rng = rng.random()
-    print((arg1, rng, arg2))
+# Example usage of _prepare_rng decorator.
+
+# Suppose a library uses a custom random state validation function, such as
+from scipy._lib._util import check_random_state
+# https://github.com/scipy/scipy/blob/94532e74b902b569bfad504866cb53720c5f4f31/scipy/_lib/_util.py#L253
+
+# Suppose a function `library_function` is defined as:
+def library_function(arg1, random_state=None, arg2=0):
+    random_state = check_random_state(random_state)
+    return random_state.random()*arg1 + arg2
+
+# We apply the decorator and change the function signature at the same time.
+# The use of `random_state` throughout the function may be replaced with `rng`,
+# or the variable may be defined as `random_state = rng`.
+@_transition_to_rng("random_state", position_num=1)
+def library_function(arg1, rng=None, arg2=0):
+    rng = check_random_state(rng)
+    return rng.random()*arg1 + arg2
```

Comment:
```suggestion
    return rng.random() * arg1 + arg2
```

Reply:
You're welcome to change it, but what is here is more consistent with PEP8 [Other Recommendations](https://peps.python.org/pep-0008/#other-recommendations).
![image](https://github.com/user-attachments/assets/abb31cb1-6239-40ba-84e2-2ef21532db1e)
![image](https://github.com/user-attachments/assets/da8a9546-64e3-4360-a5a8-a5b80262cab1)

Reply:
:eyes: I don't care very much, but reckoned it'd be consistent with what black would do. I'm confused, now, since black generates pep8 compliant code AFAIK? :thinking: